### PR TITLE
Fix LocalisableString operator== failing for non-interned strings

### PR DESCRIPTION
**Pull request by smoogipoo**
Fixes https://github.com/ppy/osu/issues/11974

Preferring `.Equals()` implementations to `operator==` for internal comparison of LocalisableString data. Note that `operator==` forwards to `string.Equals(a, b, StringComparison.Ordinal)` internally (though it does this via a by-byte-span helper so it's not immediately obvious).

I've left in `operator==` and `operator!=` because removal would be a breaking change. I can obsolete them if people think that's the right path forward, but I'm not convinced on that.

In short:
```csharp
LocalisableString localisable = "a";
string str = "a";

// Ordinal comparisons
localisable.Equals(str);
str.Equals(localisable);

// Ordinal comparisons
localisable == str;
str == localisable;

// Culture-specific comparisons
localisable.ToString().Equals(str, StringComparison.OrdinalIgnoreCase);
str.Equals(localisable.ToString(), StringComparison.OrdinalIgnoreCase);
string.Equals(localisable.ToString(), str, StringComparison.OrdinalIgnoreCase);
```

### CODE REVIEW

**Review by peppy — APPROVED** [review]

File: osu.Framework/Graphics/Sprites/SpriteText.cs [hunk 1]
```diff
@@ -97,7 +97,7 @@ public LocalisableString Text
             get => text;
             set
             {
-                if (text == value)
+                if (text.Equals(value))
```
Comment (bdach) [inline osu.Framework/Graphics/Sprites/SpriteText.cs:100]: Just to double-confirm - this change has no behavioural effect aside from saving on the allocation from the implicit conversion, right?

**Review by bdach — APPROVED** [review]
Yeah I'm OK with this. I briefly considered having the equality operators have the semantics given here (so memberwise if type matches).

File: osu.Framework/Graphics/Sprites/SpriteText.cs (hunk 1 above)
Comment (smoogipoo) [reply osu.Framework/Graphics/Sprites/SpriteText.cs:100]: Yeah this is semantically equivalent to `==`. It's left-over from a previous change where I had removed the `==` operator, but kept these changes in-case we wanted to obsolete it or something.

Comment (peppy) [reply osu.Framework/Graphics/Sprites/SpriteText.cs:100]: I guess it's preferred to use `.Equals` at osu! side as well?

Comment (smoogipoo) [reply osu.Framework/Graphics/Sprites/SpriteText.cs:100]: I'm not really fussed either way.
